File: BMICalculator/app/src/main/java/com/example/bmicalculator/MainActivity.java
/* Kalkulator BMI Kacper Kaczor
 */
package com.example.bmicalculator;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
// Deklaracja zmiennych
    private EditText height;
    private EditText weight;
    private TextView result;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        height = (EditText) findViewById(R.id.height);
        weight = (EditText) findViewById(R.id.weight);
        result = (TextView) findViewById(R.id.result);
    }
// Metoda obliczająca
    public void calBMI(View v) {
        String heightStr = height.getText().toString();
        String weightStr = weight.getText().toString();

        if (heightStr != null && !"".equals(heightStr) && weightStr != null && !"".equals(weightStr)) {
            float heightValue = Float.parseFloat(heightStr) / 100;
            float weightValue = Float.parseFloat(weightStr);

            float bmi = weightValue / (heightValue * heightValue);

            displayBMI(bmi);
        }
    }
// Wyswietlenie Wyniku
    private void displayBMI(float bmi) {
        String bmiLabel = "";
        bmiLabel = bmi + "\n\n";
        result.setText(bmiLabel);
    }

}

File: App Kotlin/BmiCalculatorKotlin/app/src/main/java/com/example/bmicalculatorkotlin/MainActivity.kt
package com.example.bmicalculatorkotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val calButton = findViewById<Button>(R.id.buttonBMI)
        val graphButton = findViewById<Button>(R.id.buttonGraph)
        val quizButton = findViewById<Button>(R.id.buttonQuiz)

        calButton.setOnClickListener {
            val intent = Intent(this, Calculator::class.java)

            startActivity(intent)
        }
        graphButton.setOnClickListener {
            val intent = Intent(this, Graph::class.java)

                startActivity(intent)
            }
        quizButton.setOnClickListener {
            val intent = Intent(this, Quiz::class.java)

            startActivity(intent)
        }
    }
}